.github/workflows/main.yml

```diff
@@ -66,9 +68,11 @@ jobs:
   unit-tests:
     uses: ./.github/workflows/unit-tests.yml
     needs:
-      - build-builder-image
+    - init
+    - build-builder-image
     with:
       collector-builder-tag: ${{ needs.build-builder-image.outputs.collector-builder-tag }}
+      include-arch-json: ${{ needs.init.outputs.include-arch-json }}
```

Comment:
This `include-arch-json` is being passed to 3 different workflows, because it is defined as a string, it is very opaque and hard to figure out what it is meant to hold when you are using it from one of these workflows. Do we have to do it this way? AFAICT, all it holds is a map that looks something like this:
```json
{
  "arch": {
    "remote": false,
    "runner": "ubuntu-24.04"
  }
}
```
If you split the workflows for local and remote like I commented previously, the remote field basically goes away and all that's left for figuring out is the runner, which only changes for arm.

Reply:
Yeah this was going to cause issues, i've removed it and the ability to specify a single arch